guix-science/packages/zotero.scm

```diff
@@ -0,0 +1,63 @@
+(define-module (guix-science packages zotero)
+  #:use-module (guix packages)
+  #:use-module ((guix licenses) #:prefix license:)
+  #:use-module (gnu packages base)
+  #:use-module (guix download)
+  #:use-module (nonguix build-system binary)
+  #:use-module (gnu packages gcc)
+  #:use-module (gnu packages gtk)
+  #:use-module (gnu packages xorg)
+  #:use-module (gnu packages glib)
+  #:use-module (ice-9 regex))
+
+    (define-public zotero (package
+      (name "zotero")
+      (version "6.0.26")
+      (source (origin
+               (method url-fetch)
+               (uri (string-append
+                     "https://download.zotero.org/client/release/" version
+                     "/Zotero-" version "_linux-x86_64.tar.bz2"))
```

Comment:
Is there a way to build it from source instead of patching and using this binary blob?

Reply:
I don't think there is, see above. All currently existing builds from the zotero website rely on the tarball, and the one from-source package on the AUR relies on 17 zotero dependencies such as separately building the note-editor, pdf-editor, translator etc. So just to get the package working, you're building 17 others, each with rely on different npm configs, among other details is a lot.